runsc/cgroup/cgroup.go

```diff
@@ -273,6 +319,8 @@ type Cgroup struct {
 	Name    string            `json:"name"`
 	Parents map[string]string `json:"parents"`
 	Own     map[string]bool   `json:"own"`
+
+	v2manager *cgroupV2Manager
```

Comment:
Let's make an interface here. The caller calls `cgroups.New` which instantiates the correct implementation based on whether cgroups V1 is mounted or not. Then the code doesn't need to check for cgroups version on every function to take a different path.

It will require some manual json marshaling and unmarshaling to handle the interface conversion through.